src/main/java/frc/robot/commands/VarShootPrime.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.classes.Structs.Range;
+import frc.robot.classes.Util;
+import frc.robot.constants.Constants;
+import frc.robot.subsystems.Elevator;
+import frc.robot.subsystems.Shooter;
+import frc.robot.subsystems.Wrist;
+import frc.robot.subsystems.chassis.PoseEstimator;
+import java.util.function.Supplier;
+import org.littletonrobotics.junction.Logger;
+
+public class VarShootPrime extends Command {
+  private Wrist wrist;
+  private Shooter shooter;
+  private Elevator elevator;
+  private PoseEstimator poseEstimator;
+  private Supplier<Translation2d> speakerTranslation;
+
+  // Translation of where the note exits in the XZ plane (side view)
+  private final Translation2d shooterXZTrans;
+
+  private final Range velRange; // Range of velocity from min distance to max distance
+  private final Range distRange; // Range of distance from min distance to max distance
+  private final double heightLengthCoeff;
+  private final double RPM_MPS;
+
+  /** Creates a new VarShootPrime. */
+  public VarShootPrime(
+      Wrist wrist,
+      Shooter shooter,
+      Elevator elevator,
+      PoseEstimator poseEstimator,
+      Translation2d shooterXZTrans,
+      Range velRange,
+      Range distRange,
+      double thetaCoeff,
+      double RPM_MPS) {
+    this.wrist = wrist;
+    this.shooter = shooter;
+    this.elevator = elevator;
+    this.poseEstimator = poseEstimator;
+    this.speakerTranslation = Constants.SPEAKER_TRANSLATION;
+    this.shooterXZTrans = shooterXZTrans;
+    this.velRange = velRange;
+    this.distRange = distRange;
+    this.heightLengthCoeff = thetaCoeff;
+    this.RPM_MPS = RPM_MPS;
+
+    addRequirements(wrist, shooter);
+  }
+
+  @Override
+  public void execute() {
+    Pose2d pose = poseEstimator.getFusedPose();
+
+    // Distance and height to speaker
+    double l = pose.getTranslation().getDistance(speakerTranslation.get()) - shooterXZTrans.getX();
```

Comment:
sparkerTranslation.get() - shooterXZTrans.getX() will always be the same value, right? You should calculate in the constructor so that it doesn't repeat the calculation each loop

Reply:
Now includes elevator height in height difference calculation

Reply:
Misread